### PR TITLE
[SFS] Refactoring sfs_file_system_v2 resource

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request
Fix warning messages
Remove redundant `goto` statement
Update schema with new arguments

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccSFSFileSystemV2_basic
--- PASS: TestAccSFSFileSystemV2_basic (56.88s)
=== RUN   TestAccSFSFileSystemV2_timeout
--- PASS: TestAccSFSFileSystemV2_timeout (36.55s)
PASS

Process finished with exit code 0


=== RUN   TestAccOTCSFSFileSystemV2_importBasic
--- PASS: TestAccOTCSFSFileSystemV2_importBasic (39.98s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go [hunk 1]
```diff
@@ -149,112 +164,112 @@ func resourceSFSFileSystemV2Create(d *schema.ResourceData, meta interface{}) err
...
 		}
-		md[key] = val
+		metadata[key] = val
 	}
-	d.Set("metadata", md)
-
-	rules, err := shares.ListAccessRights(sfsClient, d.Id()).ExtractAccessRights()
+	mErr = multierror.Append(mErr,
+		d.Set("metadata", metadata),
+	)
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go]: You can return immediately in this case, so no more API calls are done

File: opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go [hunk 2]
```diff
@@ -149,112 +164,112 @@ func resourceSFSFileSystemV2Create(d *schema.ResourceData, meta interface{}) err
...
 	}
 
 	if len(rules) > 0 {
 		rule := rules[0]
-		d.Set("share_access_id", rule.ID)
-		d.Set("access_rule_status", rule.State)
-		d.Set("access_to", rule.AccessTo)
-		d.Set("access_type", rule.AccessType)
-		d.Set("access_level", rule.AccessLevel)
+		mErr = multierror.Append(mErr,
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go]: You can simplify this conditional with 
```go
if len(rules) == 0 {
	return nil
}
```

File: opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go [hunk 3]
```diff
@@ -149,112 +164,112 @@ func resourceSFSFileSystemV2Create(d *schema.ResourceData, meta interface{}) err
...
+			return fmt.Errorf("error updating OpenTelekomCloud Share File: %s", err)
 		}
 	}
 	if d.HasChange("access_to") || d.HasChange("access_level") || d.HasChange("access_type") {
 		deleteAccessOpts := shares.DeleteAccessOpts{AccessID: d.Get("share_access_id").(string)}
-		deny := shares.DeleteAccess(sfsClient, d.Id(), deleteAccessOpts)
+		deny := shares.DeleteAccess(client, d.Id(), deleteAccessOpts)
 		if deny.Err != nil {
-			return fmt.Errorf("Error changing access rules for share file : %s", deny.Err)
+			return fmt.Errorf("error changing access rules for share file: %s", deny.Err)
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go]: In other places `ExtractErr()` is used, or smth like `if err := shares.DeleteAccess(client, d.Id(), deleteAccessOpts).Err; err != nil...`

File: opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go [hunk 4]
```diff
@@ -264,26 +279,25 @@ func resourceSFSFileSystemV2Update(d *schema.ResourceData, meta interface{}) err
...
 	if d.HasChange("size") {
-		old, newsize := d.GetChange("size")
-		if old.(int) < newsize.(int) {
-			expandOpts := shares.ExpandOpts{OSExtend: shares.OSExtendOpts{NewSize: newsize.(int)}}
-			expand := shares.Expand(sfsClient, d.Id(), expandOpts)
-			if expand.Err != nil {
-				return fmt.Errorf("Error Expanding OpenTelekomCloud Share File size: %s", expand.Err)
+		oldSize, newSize := d.GetChange("size")
+		if oldSize.(int) < newSize.(int) {
+			expandOpts := shares.ExpandOpts{OSExtend: shares.OSExtendOpts{NewSize: newSize.(int)}}
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go]: `newSize.(int)` is executed multiple times, this can be placed in a variable

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**